libusb/os/windows_winusb.c

```diff
@@ -2017,6 +2017,186 @@ static int winusb_get_device_list(struct libusb_context *ctx, struct discovered_
 	return r;
 }
 
+#ifdef _MSC_VER
+#pragma pack(push, 1)
+#endif
+
+struct string_descriptor_s {
+	UCHAR bLength;
+	UCHAR bDescriptorType;
+	uint8_t bString[252];  // UTF-16LE, must be even length
+} LIBUSB_PACKED;
+
+#ifdef _MSC_VER
+#pragma pack(pop)
+#endif
+
+struct string_descriptor_req_s {
+	USB_DESCRIPTOR_REQUEST req;
+	struct string_descriptor_s desc;
+};
+
+/*
+ * \brief Convert a UTF-16 little endian string to a UTF-8 string.
+ *
+ * \param src The input UTF-16 little endian string.
+ * \param src_length The length of src in bytes.
+ * \param dst The output UTF-8 string.
+ * \param dst_length The length of dst in bytes.
```

Comment:
Are the length variables the length of the string, or the size of the buffer? ie including/excluding null terminator?

Reply:
These are buffer sizes in bytes.  Will improve docs.